src/Features/LanguageServer/Protocol/Handler/References/FindAllReferencesHandler.cs

```diff
@@ -61,148 +48,15 @@ public FindAllReferencesHandler(IThreadingContext threadingContext, IMetadataAsS
-                else
-                {
-                    location = await ProtocolConversions.DocumentSpanToLocationAsync(documentSpan, cancellationToken).ConfigureAwait(false);
-                }
-
-                // TO-DO: The Origin property should be added once Rich-Nav is completed.
-                // https://github.com/dotnet/roslyn/issues/42847
-                var result = new LSP.VSReferenceItem
-                {
-                    ContainingMember = properties.TryGetValue(
-                        AbstractReferenceFinder.ContainingMemberInfoPropertyName, out var referenceContainingMember) ? referenceContainingMember : null,
-                    ContainingType = properties.TryGetValue(
-                        AbstractReferenceFinder.ContainingTypeInfoPropertyName, out var referenceContainingType) ? referenceContainingType : null,
-                    DefinitionId = definitionId,
-                    DefinitionText = definitionText,    // Only definitions should have a non-null DefinitionText
-                    DisplayPath = location?.Uri.LocalPath,
-                    DocumentName = documentSpan == default ? null : documentSpan.Document.Name,
-                    Id = id,
-                    Kind = symbolUsageInfo.HasValue ? ProtocolConversions.SymbolUsageInfoToReferenceKinds(symbolUsageInfo.Value) : new ReferenceKind[] { },
-                    Location = location,
-                    ProjectName = documentSpan == default ? null : documentSpan.Document.Project.Name,
-                    ResolutionStatus = ResolutionStatusKind.ConfirmedAsReference,
-                };
-
-                // Properly assigning the text property.
-                if (id == definitionId)
-                {
-                    result.Text = definitionText;
-                }
-                else if (documentSpan != default)
-                {
-                    var classifiedSpansAndHighlightSpan = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(
-                        documentSpan, cancellationToken).ConfigureAwait(false);
-                    var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
-                    var docText = await documentSpan.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
-
-                    result.Text = new ClassifiedTextElement(
-                        classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan))));
-                }
+            // Finds the references for the symbol at the specific position in the document, reporting them via streaming to the LSP client.
+            await findUsagesService.FindReferencesAsync(document, position, context).ConfigureAwait(false);
 
-                return result;
-            }
+            // The results have already been reported to the client, so we don't need to return anything here.
+            return Array.Empty<LSP.VSReferenceItem>();
```

Comment:
the line above ths is where you should call `await context.OnCompletedAsync().Conf...`